hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -654,13 +654,15 @@ pow exponent input = unsafePerformIO $ ATen.cast2 ATen.Managed.pow_tt input expo
 -- >>> dtype &&& shape $ relu (ones :: CPUTensor 'D.Float '[3,2])
 -- (Float,[3,2])
 relu ::
-  forall shape dtype device.
-  (StandardFloatingPointDTypeValidation device dtype) =>
+  forall shape dtype device t.
+  (StandardFloatingPointDTypeValidation device dtype,
+   Unnamed t, device ~ (UTDevice t), dtype ~ (UTDType t), shape ~ (UTShape t)
+  ) =>
   -- | input
-  Tensor device dtype shape ->
+  t ->
   -- | output
-  Tensor device dtype shape
-relu input = unsafePerformIO $ ATen.cast1 ATen.Managed.relu_t input
+  t
+relu input = unWrap $ unsafePerformIO $ ATen.cast1 ATen.Managed.relu_t (Wrap input)
```

Comment:
This modification does not effect current implementation like static-mnist-cnn of example-directory.